_posts/2017-09-05-planning-for-7.0.md

```diff
@@ -0,0 +1,676 @@
+
+While we want to have an easy to use tool, it turns out that many companies/developers use these "not yet JavaScript" presets all the time, and in production. "Stage 0" doesn't really set the same tone as `babel-preset-dont-use-this-stage-0`.
+
+> Ariya just made an [awesome poll](https://twitter.com/AriyaHidayat/status/833797322786025472) that explains what I'm talking about
+
+Developers don't actually know what features are in what version of JavaScript (and they shouldn't have to know). However it is a problem when we all start thinking that "features" that are actually still proposals are in the spec already.
+
+Many open source projects (including Babel still 😝), tutorials, conference talks, etc all use `stage-0`. React promotes the use of JSX, class properties (now Stage 3), object rest/spread (now Stage 3) and we all believe that it's just JavaScript because Babel compiled it for them. So maybe removing this abstraction would help people understand more about what is going on and the tradeoffs one is making when choosing to use Stage X plugins.
+
+It also seems much easier to maintain your own preset than to have to update the Stage preset.
+
+> I often see people go "I want object rest, and that's stage 2, so I enabled stage 2". They now have a load of other experimental features enabled they might not know about and probably don't need.
+> Also, as stages change over time then people who aren't using shrinkwrap or yarn will get new features appearing, possibly without their knowledge. If a feature is canned they might even get one vanishing. @glenjamin
+
+### Using npm Scoped Packages
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Thoughts on <a href="https://twitter.com/babeljs">@babeljs</a> using npm scoped packages for 7.0?</p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/821551189166878722">January 18, 2017</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Seems like most who understood what scoped packages are were in favor?
+
+Pros
+
+- Don't have to worry about getting a certain package name (the reason why this was brought up in the first place).
+
+> Many package names have been taken (preset-es2016, preset-es2017, 2020, 2040, etc). Can always ask to transfer but not always easy to do and can lead to users believing certain packages are official due to the naming.
+
+Cons
+
+- We need to migrate to new syntax
+- Still unsupported on certain non-npm tools (lock-in)
+- No download counts unless we alias back to old names
+
+Sounds like we may want to defer, and in the very least it's not a breaking change given it's a name change.
+
+### `external-helpers`, `transform-runtime`, `babel-polyfill`
+
+> "regeneratorRuntime is not defined" - reported all the time.
```

Comment:
I believe the situation could be improved with this PR merged in - https://github.com/facebook/regenerator/pull/307

It refactors the `regenerator-runtime`'s code to es6 modules and it contains a change to the `regenerator-transform` which leverages the former and adds an import statement to the `regenerator-runtime` (similar thing is done by `transform-runtime` nowadays). 

With that in place I dont know how one would end up with "regeneratorRuntime is not defined" - it just should be defined at all times.